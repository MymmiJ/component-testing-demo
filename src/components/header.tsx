export const Header = ({ pageTitle }: { pageTitle: string}) => <header data-testid="header">
  <h1>{pageTitle}</h1>
</header>;
