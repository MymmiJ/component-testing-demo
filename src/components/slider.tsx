import { ChangeEvent, useCallback, useState } from 'react'

const Slider = ({min = 0, max = 256}) => {
  const [number, setNumber] = useState(0)
  const [numberEntered, setNumberEntered] = useState(false)
  const handleChange = useCallback((event: ChangeEvent<HTMLInputElement>) => {
    setNumber(Number(event.target.value))
    setNumberEntered(true)
  }, [setNumber, setNumberEntered]);
  const isValid = !numberEntered || (number >= min && number <= max)
  return (
    <div>
      <label htmlFor="spinbutton">Slider</label>
      <input
        id="spinbutton"
        role="spinbutton"
        aria-valuenow={number}
        type="number"
        value={number}
        onChange={handleChange}
      />
      {isValid ? null : <div role="alert">Error: Invalid number</div>}
    </div>
  )
}

export { Slider }
