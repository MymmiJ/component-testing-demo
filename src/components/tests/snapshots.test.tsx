import { render, screen } from '@testing-library/react';
import { Header } from '../header';
import { Slider } from '../slider';

describe('Snapshot tests', () => {
  it('has not changed the rendering of the header element', () => {
    render(<Header pageTitle="Example" />);
    const header = screen.getByTestId('header');
    expect(header).toMatchInlineSnapshot(`
<header
  data-testid="header"
>
  <h1>
    Example
  </h1>
</header>
`);
  });
  it('has not changed the rendering of the slider element', () => {
    render(<Slider min={20} max={50} />);
    const slider = screen.getByRole('spinbutton');
    expect(slider).toMatchSnapshot();
  })
});