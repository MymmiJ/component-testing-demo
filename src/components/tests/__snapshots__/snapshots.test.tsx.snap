// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot tests has not changed the rendering of the slider element 1`] = `
<input
  aria-valuenow="0"
  id="spinbutton"
  role="spinbutton"
  type="number"
  value="0"
/>
`;
